pkg/util/log/log.go

```diff
@@ -534,6 +534,11 @@ func Tracef(format string, params ...interface{}) {
 	logFormat(seelog.TraceLvl, func() { Tracef(format, params...) }, Logger.tracef, format, params...)
 }
 
+func TracefStackDepth(format string, depth int, params ...interface{}) {
+	m := fmt.Sprintf(format, params...)
+	logContext(seelog.TraceLvl, func() { Tracec(m, nil) }, Logger.trace, m, depth, nil)
+}
```

Comment:
`Sprintf` is very costly cpu wise so we don't want to compute it before checking the log level.
We actually had issues with that recently in the log component, see https://github.com/DataDog/datadog-agent/pull/17876.